numpy/lib/type_check.py

```diff
@@ -271,6 +279,11 @@ def isreal(x):
     --------
     >>> np.isreal([1+1j, 1+0j, 4.5, 3, 2, 2j])
     array([False,  True,  True,  True,  True, False])
+    
+    Returns True for all elements in input array which are not of complex type. 
+
+    >>> np.isreal([None, False, "imaginary", complex])
+    array([True, True, True, True])
```

Comment:
When tested, it returns False instead of an array in the following cases.
```
>>> np.isreal([2j, "a"])
/.../lib/python3.6/site-packages/numpy/lib/type_check.py:277: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison
  return imag(x) == 0
False

```

Reply:
That happens for string arrays, on current master, it should run into two future warnings, because the original input is already a mix of string and complex... Maybe fine to ignore.

Reply:
Maybe we should keep the `Notes` section as it is, and move the `.. note::` directive down there with more details that the function does not work for string arrays, and for `dtype=object` currently always returns True even if the content is complex.

If we want an example, maybe include `np.array([1+2j, ...], dtype=object)` (a complex number and the `dtype=object` explicitly)?

Reply:
@cuspymd, @seberg, thanks for the comments. i've made fixes by moving the `.. note::` directive down to the examples section as well as including the example for string array and mentioning that the function doesn't work, as well as the array with python objects returning all `True` even if it consists complex numbers. let me know if more clarifications can be added!